python/pyspark/tests/test_worker.py

```diff
@@ -150,6 +151,28 @@ def test_with_different_versions_of_python(self):
         finally:
             self.sc.pythonVer = version
 
+    def test_python_exception_non_hanging(self):
+        """
+        SPARK-21045: exceptions with no ascii encoding shall not hanging PySpark.
+        """
+        def f():
+            raise Exception("exception with 中 and \xd6\xd0")
+
+        def run():
+            self.sc.parallelize([1]).map(lambda x: f()).count()
```

Comment:
Can we actually just run this code, catch the exception and check the error message simply?

Reply:
After the patch to worker, we may simply run this code in the main thread. But it will hanging 
forever without the patch. 

I am prefer this way to indicate that we actually fix the hanging.

Reply:
Let's just test directly without a thread. Actually indefinitely hanging is another issue that the exception thrown during exception handling is not dealt with.